### PR TITLE
fix: display IRI metadata

### PR DESCRIPTION
**Pull request by mhagel**
## Description

Closes: regen-network/regen-registry#1322
Closes: regen-network/regen-web#1366

- Fix: display IRI-resolved metadata for onchain projects (previously it was showing the meta from db project.metadata)
- Add `MsgUpdateProjectMetadata`, `MsgUpdateProjectAdmin`, and `MsgBridge` to Activity page.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

All 3 project variations should look right (especially AdditionalMetadata)
1. https://deploy-preview-1444--regen-registry.netlify.app/projects/C45-013
2. https://deploy-preview-1444--regen-registry.netlify.app/projects/C01-001
3. https://deploy-preview-1444--regen-registry.netlify.app/projects/wilmot

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx [hunk 1]
```diff
@@ -69,16 +75,12 @@ function ProjectTopSection({
   const apiServerUrl = process.env.REACT_APP_API_URI;
 
   const project = data?.projectByOnChainId || data?.projectByHandle; // TODO: eventually just projectByOnChainId
-  const metadata = project?.metadata; // TODO: this is from postgres metadata - needs to be from metadata resolver instead
```
Comment (mhagel) [inline web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:72]: This was the key pain point. Even though we fetched the IRI metadata in the ProjectDetails component, this was re-assigning the metadata from the postgres table (similar, but not reflective of on-chain metadata updates).

File: web-registry/src/generated/json-ld/project-metadata.ts [hunk 2]
```diff
@@ -25,7 +27,7 @@ export interface ProjectMetadataLD {
   'regen:galleryPhotos': UrlList;
   'regen:previewPhoto': UrlType;
   'regen:videoURL': UrlType;
-  'regen:glanceText': TypeValue[];
+  'regen:glanceText': any;
```
Comment (mhagel) [inline web-registry/src/generated/json-ld/project-metadata.ts:30]: This is only in legacy projects. Taking the easy route here.

File: web-registry/src/components/atoms/Link.tsx [hunk 3]
```diff
@@ -17,7 +17,9 @@ export const Link: React.FC<LinkProps> = ({
   target,
   ...linkProps
 }) => {
-  if (!href) return <Box {...linkProps}>{children}</Box>;
+  if (!href || typeof href !== 'string') {
```
Comment (mhagel) [inline web-registry/src/components/atoms/Link.tsx:20]: handling edge case - unknown type

File: web-registry/src/components/atoms/LinkWithArrow.tsx [hunk 4]
```diff
@@ -22,7 +22,11 @@ const LinkWithArrow: React.FC<LinkWithArrowProps> = ({
   className,
   target,
 }) => {
-  const defaultTarget = href?.startsWith('/') ? '_self' : '_blank';
+  const defaultTarget =
+    !!href && typeof href === 'string' && href.startsWith('/')
```
Comment (mhagel) [inline web-registry/src/components/atoms/LinkWithArrow.tsx:26]: handling edge case - unknown type

File: web-registry/src/generated/json-ld/project-metadata.ts [hunk 5]
```diff
@@ -16,6 +16,8 @@ export interface ProjectMetadataLD {
   'regen:landOwner': ProjectStakeholder;
   'regen:projectOriginator': ProjectStakeholder;
   'regen:offsetGenerationMethod': string;
+  // TODO: not sure why regen:offsetGenerationMethod isn't compacting as expected so added this:
+  'http://regen.network/offsetGenerationMethod'?: string;
```
Comment (mhagel) [inline web-registry/src/generated/json-ld/project-metadata.ts:20]: @wgwz Not sure why I couldn't get this to compact as expected.

File: web-registry/src/lib/ecocredit/constants.ts [hunk 6]
```diff
@@ -131,4 +134,16 @@ export const ECOCREDIT_MESSAGE_TYPES = {
     message: `/${MsgUpdateSellOrders.$type}`,
     readable: 'update sell order',
   },
+  BRIDGE: {
```
Comment (mhagel) [inline web-registry/src/lib/ecocredit/constants.ts:137]: This is coming soon, so just adding it now

**Review by flagrede — APPROVED** [review]
Looking good! Will close my other PR on the start/end date for CFC.

File: web-registry/src/generated/json-ld/project-metadata.ts (hunk 5 above)
Comment (wgwz) [reply web-registry/src/generated/json-ld/project-metadata.ts:20]: Glad to take a look at this with you another time if it's not blocking us at the moment. But if you do want to spend some time on it, just let me know

**Review by wgwz — APPROVED** [review]
Approving but i'm glad to take a look at the offset generation thing with you if you want. Either way, maybe we should open a separate ticket for that? If you do open the ticket it'd be great to include what you're expecting the compacted metadata for that field to look like, versus what it actually looks like, and then if possible a project ID reference

**Comment by mhagel** [comment]
New issue for `offsetGenerationMethod`: regen-network/regen-registry#1327

**Pull request merged**